### PR TITLE
feat: save data-theme state in localstorage

### PR DESCRIPTION
**Pull request by yin1999**
# What does this PR do?

Fixes: #848. Using localstorage to store the theme state.

### CODE REVIEW

File: web/writing.html [hunk 1]
```diff
@@ -797,7 +797,7 @@ <h5 class="portlet__head">Webhook</h5>
   </script>
   <script>
     window.addEventListener('load', () => {
-      if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+      if (localStorage.getItem('theme') === 'dark' || window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (jeessy2) [inline web/writing.html:800]: 判斷有點問題
`localStorage.getItem('theme') === 'dark'  || (localStorage.getItem('theme') === null && window.matchMedia('(prefers-color-scheme: dark)').matches)`

**Comment by jeessy2** [comment]
如果我是暗色模式，調整為了light模式
`localStorage.getItem('theme') === 'dark' || window.matchMedia('(prefers-color-scheme: dark)').matches` 這段為true，本來我設置的為light，自動切換成dark了

**Comment by yin1999** [comment]
> 如果我是暗色模式，調整為了light模式 `localStorage.getItem('theme') === 'dark' || window.matchMedia('(prefers-color-scheme: dark)').matches` 這段為true，本來我設置的為light，自動切換成dark了

想调整一下 CSS 选择器的，略微有些麻烦。后面有时间再尝试了。

**Comment by jeessy2** [comment]
ok, thanks

**Pull request merged**